diagnostics-extension/src/services/diagnostics.ts

```diff
@@ -41,28 +44,37 @@ const mapRoutineNameToMethod = (name: DiagnosticsRoutineName) => {
 export class FakeDiagnosticsService implements DiagnosticsService {
   private _activeRoutines: { [key: number]: Routine } = {};
 
+  private _fetchRoutineById = (id: number) => {
+    if (!(id in this._activeRoutines)) {
+      throw ResponseErrorInfoMessage.InvalidDiagnosticsRoutineId;
+    }
+    return this._activeRoutines[id];
+  };
+
   runRoutine = async (
     name: DiagnosticsRoutineName,
     params?: DiagnosticsParams
   ): Promise<number> => {
     params && console.log('Recieved params', params);
     const dpslRoutineMethod = mapRoutineNameToMethod(name);
-    if (!dpslRoutineMethod) return -1;
-    const routine = dpslRoutineMethod();
+    if (!dpslRoutineMethod) {
+      throw ResponseErrorInfoMessage.InvalidDiagnosticsRoutineName;
+    }
+    const routine = await dpslRoutineMethod();
     this._activeRoutines[routine.id] = routine;
     return routine.id;
```

Comment:
I'd rather not to use "await" here, this function should be executed asynchronously. It is better to specify what it returns in the transformer function "then".

return dpslRoutineMethod().then((routine) => {
  this._activeRoutines[routine.id] = routine;
  return routine.id;
});